parsl/channels/local/local.py

```diff
@@ -65,7 +65,8 @@ def execute_wait(self, cmd, walltime=None, envs={}):
                 stderr=subprocess.PIPE,
                 cwd=self.userhome,
                 env=current_env,
-                shell=True
+                shell=True,
+                preexec_fn=os.setpgrp
```

Comment:
The docs indicate that `preexec_fn` mechanism not threadsafe and recommends using the `start_new_session=True` instead. Perhaps that isn't important right now, since making the change also requires updating the kill mechanism to use sessions instead.

Reply:
This should have around the same level (good or bad) of thread safety as using it in execute_no_wait. I'll open an issue noting this.

Reply:
Agreed. Merging now.